Hotfix/amlogs credentials

JasonConger opened the pull request.

JasonConger committed:
Create unique credential for each metric input

Create a credential based on the input name in order to support
multiple metric inputs.  Also, this hotfix will migrate the existing
credentials to new credentials to support upgrades.

JasonConger committed:
Create unique credential for each log input

Create a credential based on the input name in order to support
multiple activity and diagnostic log inputs. Also, this hotfix will
migrate the existing credentials to new credentials to support upgrades.

sebastus commented:
@JasonConger Did you test what happens when using MSI creds? (leaving appid, key, tenant id blank)

JasonConger committed:
Check for MSI use

App ID and key will be blank if using MSI creds, so only try to update
the storage passwords if not using MSI.

JasonConger commented:
@sebastus good point about MSI creds.  I moved some code around in the metrics input to only do the password dance update if app_id and app_key are defined.  The logs input didn't need this change as the code I added was already in a conditional where this will not run for MSI creds.

sebastus reviewed (changes requested):
That's all I found. Thanks again for the contribution. :)
